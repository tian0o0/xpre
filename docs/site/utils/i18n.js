// component mixin
import { get } from '../../../src/utils';
import { camelize } from '../../../src/utils/format/string';

export default {
  computed: {
    $t() {
      const { name } = this.$options;
      const prefix = name ? camelize(name) + '.' : '';
      const messages = this.$vantMessages[this.$vantLang];

      return (path, ...args) => {
        const message = get(messages, prefix + path) || get(messages, path);
        return typeof message === 'function' ? message(...args) : message;
      };
    }
  }
};
