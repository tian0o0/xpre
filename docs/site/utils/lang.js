import Locale from '../../../src/components/locale';
import zhCN from '../../../src/components/locale/lang/zh-CN';
import enUS from '../../../src/components/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Xpre - Exploring Endless Possibilities',
    messages: enUS
  },
  'zh-CN': {
    title: 'Xpre | 一个差不多的组件库',
    messages: zhCN
  }
};

let currentLang = '';

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const { hash } = location;

  for (let i = 0; i < langs.length; i++) {
    if (hash.indexOf(langs[i]) !== -1) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('VANT_LANGUAGE') || navigator.language || 'en-US';
  return userLang.indexOf('zh-') !== -1 ? 'zh-CN' : 'en-US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('VANT_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}

setLang(getDefaultLang());
