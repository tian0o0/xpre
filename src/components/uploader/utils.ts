export function toArray<T>(item: T | T[]): T[] {
  if (Array.isArray(item)) {
    return item;
  }

  return [item];
}

export function readFile(file: File, resultType: string) {
  return new Promise(resolve => {
    const reader = new FileReader();

    reader.onload = event => {
      resolve((event.target as FileReader).result);
    };

    if (resultType === 'dataUrl') {
      reader.readAsDataURL(file);
    } else if (resultType === 'text') {
      reader.readAsText(file);
    }
  });
}

export function isOversize(files: File | File[], maxSize: number): boolean {
  return toArray(files).some(file => file.size > maxSize);
}

export type FileListItem = {
  url?: string;
  file?: File;
  content?: string; // dataUrl
};

const IMAGE_EXT = ['jpeg', 'jpg', 'gif', 'png', 'svg'];

export function isImageUrl(url: string): boolean {
  return IMAGE_EXT.some(ext => url.indexOf(`.${ext}`) !== -1);
}

export function isImageFile(item: FileListItem): boolean {
  if (item.file && item.file.type) {
    return item.file.type.indexOf('image') === 0;
  }

  if (item.url) {
    return isImageUrl(item.url);
  }

  if (item.content) {
    return item.content.indexOf('data:image') === 0;
  }

  return false;
}
