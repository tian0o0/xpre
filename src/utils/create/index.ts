import { createBEM, Bem } from './bem';
import { createComponent, Component } from './component';
import { createI18N, Translate } from './i18n';

type CreateNamespaceReturn = [
  Component,
  Bem,
  Translate
];

export function createNamespace(name: string): CreateNamespaceReturn {
  name = 'x-' + name;
  return [createComponent(name), createBEM(name), createI18N(name)];
}
