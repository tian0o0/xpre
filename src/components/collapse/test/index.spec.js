import Collapse from '..';
import CollapseItem from '../../collapse-item';
import { later, mount } from '../../../../test/utils';

const component = {
  template: `
  <collapse v-model="active" :accordion="accordion" :border="border">
    <collapse-item title="a" name="first">content</collapse-item>
    <collapse-item title="b">content</collapse-item>
    <collapse-item title="c">content</collapse-item>
  </collapse>
  `,
  components: {
    Collapse,
    CollapseItem
  },
  props: {
    accordion: Boolean,
    border: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {
      active: this.accordion ? '' : []
    };
  }
};

test('basic mode', async () => {
  const wrapper = mount(component);

  const titles = wrapper.findAll('.x-collapse-item__title');
  titles.at(0).trigger('click');
  expect(wrapper.vm.active).toEqual(['first']);

  await later();
  titles.at(1).trigger('click');
  expect(wrapper.vm.active).toEqual(['first', 1]);

  await later();
  titles.at(0).trigger('click');
  expect(wrapper.vm.active).toEqual([1]);

  wrapper.destroy();
});

test('accordion', async () => {
  const wrapper = mount(component, {
    propsData: {
      accordion: true
    }
  });

  const titles = wrapper.findAll('.x-collapse-item__title');
  titles.at(0).trigger('click');
  expect(wrapper.vm.active).toEqual('first');

  titles.at(1).trigger('click');
  expect(wrapper.vm.active).toEqual(1);

  await later();
  titles.at(0).trigger('click');
  expect(wrapper.vm.active).toEqual('first');

  titles.at(0).trigger('click');
  expect(wrapper.vm.active).toEqual('');
});

test('render collapse-item slot', () => {
  const wrapper = mount({
    template: `
      <collapse v-model="active">
        <collapse-item>
          <template v-slot:title>this is title</template>
          <template v-slot:value>this is value</template>
          <template v-slot:icon>this is icon</template>
          <template v-slot:right-icon>this is right icon</template>
        </collapse-item>
      </collapse>
      `,
    components: {
      Collapse,
      CollapseItem
    },
    data() {
      return {
        active: []
      };
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('disable border', () => {
  const wrapper = mount(component, {
    propsData: {
      border: false
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('lazy render collapse content', async () => {
  const wrapper = mount({
    template: `
      <collapse v-model="active">
        <collapse-item title="a" name="first" style="padding: 0;">content</collapse-item>
        <collapse-item title="b" style="padding: 0;">{{ content }}</collapse-item>
      </collapse>
    `,
    components: {
      Collapse,
      CollapseItem
    },
    data() {
      return {
        content: '',
        active: []
      };
    }
  });

  const titles = wrapper.findAll('.x-collapse-item__title');

  titles.at(1).trigger('click');
  wrapper.vm.content = 'content';
  expect(wrapper).toMatchSnapshot();
});
