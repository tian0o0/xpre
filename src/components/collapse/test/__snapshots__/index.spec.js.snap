// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disable border 1`] = `
<div class="x-collapse">
  <div class="x-collapse-item">
    <div role="button" tabindex="0" aria-expanded="false" class="x-cell x-cell--clickable x-collapse-item__title">
      <div class="x-cell__title"><span>a</span></div><i class="x-icon x-icon-arrow x-cell__right-icon"></i>
    </div>
  </div>
  <div class="x-collapse-item x-hairline--top">
    <div role="button" tabindex="0" aria-expanded="false" class="x-cell x-cell--clickable x-collapse-item__title">
      <div class="x-cell__title"><span>b</span></div><i class="x-icon x-icon-arrow x-cell__right-icon"></i>
    </div>
  </div>
  <div class="x-collapse-item x-hairline--top">
    <div role="button" tabindex="0" aria-expanded="false" class="x-cell x-cell--clickable x-collapse-item__title">
      <div class="x-cell__title"><span>c</span></div><i class="x-icon x-icon-arrow x-cell__right-icon"></i>
    </div>
  </div>
</div>
`;

exports[`lazy render collapse content 1`] = `
<div class="x-collapse x-hairline--top-bottom">
  <div class="x-collapse-item" style="padding: 0px;">
    <div role="button" tabindex="0" aria-expanded="false" class="x-cell x-cell--clickable x-collapse-item__title">
      <div class="x-cell__title"><span>a</span></div><i class="x-icon x-icon-arrow x-cell__right-icon"></i>
    </div>
  </div>
  <div class="x-collapse-item x-hairline--top" style="padding: 0px;">
    <div role="button" tabindex="0" aria-expanded="true" class="x-cell x-cell--clickable x-collapse-item__title x-collapse-item__title--expanded">
      <div class="x-cell__title"><span>b</span></div><i class="x-icon x-icon-arrow x-cell__right-icon"></i>
    </div>
    <div class="x-collapse-item__wrapper">
      <div class="x-collapse-item__content">content</div>
    </div>
  </div>
</div>
`;

exports[`render collapse-item slot 1`] = `
<div class="x-collapse x-hairline--top-bottom">
  <div class="x-collapse-item">
    <div role="button" tabindex="0" aria-expanded="false" class="x-cell x-cell--clickable x-collapse-item__title">this is icon<div class="x-cell__title">this is title</div>
      <div class="x-cell__value">this is value</div>this is right icon
    </div>
  </div>
</div>
`;
